donation/views.py

```diff
@@ -17,52 +32,167 @@ class CustomPageNumberPagination(PageNumberPagination):
-                        Q(is_approved=False) &
-                        Q(is_rejected=False)
-                    ).order_by('priority')
-                else:
-                    queryset = DonationRequest.objects.filter(
-                        Q(is_approved=False) &
-                        Q(is_rejected=False)
-                    ).order_by('-priority')
-
-            else:
-                queryset = DonationRequest.objects.filter(
-                    Q(is_approved=False) &
-                    Q(is_rejected=False)).order_by('-time')
-
+            queryset = DonationRequest.objects.filter(
+                Q(is_approved=False) &
+                Q(is_rejected=False) &
+                Q(in_progress=False)
+            )
+            return queryset
         else:
             queryset = DonationRequest.objects.filter(
                 ~Q(created_by=self.request.user) &
-                Q(is_approved=True)).order_by('-time')
+                Q(is_approved=True) &
+                Q(in_progress=False)).order_by('-time')
         return queryset
 
 
 class UserRequestsView(generics.ListAPIView):
+    pagination_class = CustomPageNumberPagination
+    serializer_class = DonationUserSerializer
+
     def get_queryset(self):
         return DonationRequest.objects.filter(created_by=self.request.user)
 
-    def list(self, request):
-        queryset = self.get_queryset()
-        serializer = DonationSerializer(queryset, many=True)
-        return Response(serializer.data)
-
 
 class DonationUpdateDestoryView(generics.RetrieveUpdateDestroyAPIView):
     permission_classes = [
         IsUserAuthorized,
     ]
     queryset = DonationRequest.objects.all()
-    serializer_class = DonationSerializer
+    serializer_class = DonationUserSerializer
+
+
+class DonationAdminActionsView(generics.UpdateAPIView):
+    permission_classes = [
+        permissions.IsAdminUser,
+    ]
+    serializer_class = BaseSerializer
+    queryset = DonationRequest.objects.all()
+
+
+class DonationInProgressActionView(generics.UpdateAPIView):
+    serializer_class = DonationInProgressActionSerializer
+    queryset = DonationRequest.objects.all()
+
+    @staticmethod
+    def send_mail(template_id, sender, recipient, data_dict):
+        print('here')
+        sendGridClient = sendgrid.SendGridAPIClient(SENDGRID_API_KEY)
+        mail = Mail()
+        mail.template_id = template_id
+
+        mail.from_email = Email(sender)
+        personalization = Personalization()
+        personalization.add_to(Email(recipient))
+        personalization.dynamic_template_data = data_dict
+        mail.add_personalization(personalization)
+
+        try:
+            response = sendGridClient.client.mail.send.post(
+                request_body=mail.get())
+            print(response)
+        except exceptions.BadRequestsError as e:
+            print("INSIDE")
+            print(e.body)
+
+    @staticmethod
+    @receiver(signals.post_save, sender=DonationRequest)
```

Comment:
signals shouldn't be in views, there should be a separate file to store these handlers